noir/noir-repo/aztec_macros/src/transforms/compute_note_hash_and_nullifier.rs

```diff
@@ -59,13 +62,86 @@ pub fn inject_compute_note_hash_and_nullifier(
             return Ok(());
         }
 
+        let traits: Vec<_> = collect_traits(context);
+
         // In order to implement compute_note_hash_and_nullifier, we need to know all of the different note types the
         // contract might use. These are the types that are marked as #[aztec(note)].
-        let note_types =
-            fetch_notes(context).iter().map(|(path, _)| path.to_string()).collect::<Vec<_>>();
+        let mut note_types = vec![];
+        let mut note_lengths = vec![];
+
+        let max_note_length_const = context
```

Comment:
I'd consider moving this into a `get_numeric_constant(str) -> u128` function, this fn is already quite large as is and this is a nice standalone block

Reply:
Extracted to utility fn in `hir_utils`